lib/websocket.js

```diff
@@ -4,26 +4,33 @@ const { SubscriptionClient } = require('subscriptions-transport-ws');
 const ws = require('ws');
 const gql = require('graphql-tag');
 const log = require('./log');
+const Config = require('../src/Config');
 
-const getWsClient = function(wsurl, apiKey) {
+const createConnectionParams = function () {
+  return {
+    headers: {
+      'razee-org-key': Config.orgKey
+    }
+  };
+}
+
+const getWsClient = function (wsurl) {
   log.debug(`creating websock client to ${wsurl}`);
   const client = new SubscriptionClient(
-    wsurl, {
+    wsurl,
+    {
       reconnect: true,
       connectionCallback: () => { log.info(`websocket connection established with ${wsurl}`); },
-      'connectionParams': {
-        headers: {
-          'razee-org-key': apiKey
-        }
-      },
-    }, ws
+      connectionParams: createConnectionParams
```

Comment:
If the socket is closed by the server (because api key is no longer valid, for example), the client will try to reconnect with up-to-date api key. It's not necessary to have the client preemptively close the current socket when api key change is detected. `createConnectionParams` will return up-to-date api key when the client tries to reconnect to the server.